### ISSUE TITLE
Better error message for "Cannot find name" when looking for a type

### ISSUE THREAD

**Issue by a8775** [microsoft/TypeScript#7900]
**TypeScript Version:** 
1.8.7

**Code**

tsconfig.json

``` json
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6",
        "declaration": true
    },
    "files": [
        "main.ts",
        "mytypes.ts"
    ]
}
```

main.ts

``` ts
import {MyClassAlias} from './mytypes';

let a: MyClassAlias = new MyClassAlias();
```

mytypes.ts

``` ts
export class MyClass {
    public a: number;
}

export type MyClassAlias = MyClass;
```

**Expected behavior:**
Object created using alias type.

**Actual behavior:** 
Compilation error:

```
main.ts(3,27): error TS2304: Cannot find name 'MyClassAlias'.

```

in `vscode` (0.10.11) syntax is marked as error as well:

![snip_20160406234306](https://cloud.githubusercontent.com/assets/14092227/14333957/8ed73bb0-fc51-11e5-8e7d-459b05b0f03b.png)

but IntelliSense seems to work:

![snip_20160406234539](https://cloud.githubusercontent.com/assets/14092227/14333989/c42ea8c0-fc51-11e5-89eb-04dc58dff1ad.png)

**Comment by jwbay** [comment]
a type alias is a compile-time construct. there's no javascript emitted for it, so you can't `new` it up.

**Comment by RyanCavanaugh** [comment]
This is working as intended (as noted, `type` is only a type; use a `const` + `type` pair if you want both definitions), but we should give a better error message when this happens like we do in other cases where symbol resolution fails in one namespace but would succeed in another.

**Comment by nplott** [comment]
What do you guys think of
```
 Cannot instantiate type 'MyClassAlias'. Operator new cannot be used with alias types.
```

**Comment by RyanCavanaugh** [comment]
This error message should be general to all cases where you refer to an identifier in a value position which only resolves to something in the type namespace. In other words:
```ts
interface JustAType { }
var x = JustAType; // Error: __(fill in the blank)__
```

I was thinking something like
 > Cannot find name 'JustAType'. A type exists with this name, but no value.

**Comment by mhegazy** [comment]
thanks @jwbay!
